src/sfctl/custom_deployment.py

```diff
@@ -0,0 +1,131 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# -----------------------------------------------------------------------------
+
+"""Custom commands for managing Service Fabric Mesh resources"""
+
+from __future__ import print_function
+
+import enum
+import json
+import os
+import shutil
+from knack.util import CLIError
+from sfmergeutility import SFMergeUtility
+
+class ResourceType(enum.Enum):
+    """ Defines the valid yaml resource types
+        which are parseable by CLI
+    """
+    application = 1
+    volume = 2
+    network = 3
+    secret = 4
+    secretValue = 5
+    gateway = 6
+
+def get_resource_type(file_name):
+    """ Gets the resource type from the file name
+    :param str file_name: Path of the file
+    """
+    file_name = os.path.basename(file_name)
+    file_name_splitted = file_name.split('_')
+    if len(file_name_splitted) < 3:
+        raise CLIError('Invalid resource file name %s. The file name should be of format id_resourcetype_resourcename.json' %(file_name)) # pylint: disable=line-too-long
+    resource_type = file_name_splitted[1]
+    try:
+        return ResourceType[resource_type]
+    except:
+        raise CLIError('The resource type %s is unknown' %(resource_type))
+
+def get_resource_name(file_name):
+    """ Gets resource name from the file name
+    :param str file_name: Path of the file
+    """
+    file_name = os.path.basename(file_name)
+    file_name_splitted = file_name.split('_')
+    if len(file_name_splitted) < 3:
+        raise CLIError('Invalid resource file name %s. The file name should be of format id_resourcetype_resourcename.json' %(file_name)) # pylint: disable=line-too-long
+    file_name_with_extension = file_name_splitted[2]

```

Comment:
should this be called resource_name_with_extension instead?

Reply:
took your suggestion to the merge utility code as this is no longer here

Reply:
took your feedback from and applied it to the code at it's newest location